build-in-container/run.sh

```diff
@@ -0,0 +1,52 @@
+#! /bin/bash
+
+help() {
+    echo "------------ Mariner Build-in-Container ------------"
+    echo "
+    The run.sh script presents these options
+    -t      creates container image
+    -b      creates container, 
+            builds the specs under $(pwd)/SPECS/ 
+            and places the output under $(pwd)/out/
+    -i      create an interactive Mariner build container
+    -c      cleans up the current workspace, container images and instances
+    --help  shows help on usage
+    "
+    echo "----------------------------------------------------"
+}
+
+create_container() {
+    echo "Creating Container Image"
+    source create-build-container.sh
+}
+
+build_mariner() {
+    echo "Creating Mariner Build Container and building Mariner SPECS"
+    source mariner-docker-run.sh build
```

Comment:
conditionally sourcing other files in a function seems a bit odd, its not a pattern I've ever seen before. (The pattern I've seen is sourcing everything you need once, and using functions inside the sourced files as needed)

Reply:
I'm calling the same file to start a docker container, but with different args (interactive v/s build mode)